proto_parser/src/proto_value_test.cc

```diff
@@ -22,8 +22,166 @@
 #include <google/protobuf/util/message_differencer.h>
 
 namespace wireless_android_play_analytics{
+ 
+const std::string text_proto = R"pb(
+# comment 1
+int32_field: 1
 
-// TODO(alexanderlin): Implement.
+# comment 2
+field_nested_message { # comment 3
+  int64_field: 100
+}
+
+field_nested_message {
+  # comment 4
+  # comment 5
+  int64_field: 100
+}
+
+
+bool_field: true
+bool_field: false # comment 6
+)pb";
+
+class ProtoValueTest : public ::testing::Test {
+ protected:
+  
+  virtual void SetUp() {
+    message = ProtoValue::Create(text_proto, test);
+    message_val = dynamic_cast<MessageValue*>(message.get());
+  }
+
+  // virtual void TearDown() {}
+
+  TestProto test;
+  std::unique_ptr<ProtoValue> message;
+  MessageValue* message_val;
+};
+
+TEST_F(ProtoValueTest, OuterProtoTest) {
+  const std::vector<std::unique_ptr<ProtoValue>>& fields = 
+      message_val->GetFields();
+  EXPECT_EQ("", message_val->GetName());
```

Comment:
nit: EXPECT_EQ(message_val->GetName(), ""). we should put the expected result after comma.
same for all the rest.